lib/matplotlib/transforms.py

```diff
@@ -2975,3 +2976,26 @@ def offset_copy(trans, fig=None, x=0.0, y=0.0, units='inches'):
         y /= 72.0
     # Default units are 'inches'
     return trans + ScaledTranslation(x, y, fig.dpi_scale_trans)
+
+
+class ScaledRotation(Affine2DBase):
+    """
+    A transformation that applies offset and direction
+    based on *trans_shift*.
+    """
+    def __init__(self, theta, trans_shift, is_radial):
+        super().__init__()
+        self._theta = theta
+        self._trans_shift = trans_shift
+        self._is_radial = is_radial
+        self._mtx = np.identity(3)
+
+    def get_matrix(self):
+        if self._invalid:
+            transformed_coords = self._trans_shift.transform([[self._theta, 0]])[0]
+            adjusted_theta = transformed_coords[0]
+            if self._is_radial:
+                adjusted_theta -= np.pi / 2
```

Comment:
Just to confirm, you can't apply this adjustment to the _input_ to `ScaledRotation` (and thus not have the `is_radial` parameter), right?

Reply:
That's correct. I tried applying the adjustment to the input before the transform, like this:

```python 
for theta, r in zip(l.get_xdata(), l.get_ydata()):
                        if axis == 'y':
                            theta -= np.pi/2
                        rotation = ScaledRotation(
                            theta=theta,
                            trans_shift=trans_shift)
                        ms = mmarkers.MarkerStyle(marker=marker,
                                                  transform=rotation)
                        self.add_line(mlines.Line2D([theta], [r], marker=ms,
                                                    **eb_cap_style))
```

```python 
class ScaledRotation(Affine2DBase):
    """
    A transformation that applies offset and direction
    based on *trans_shift*.
    """
    def __init__(self, theta, trans_shift):
        super().__init__()
        self._theta = theta
        self._trans_shift = trans_shift
        self._mtx = None

    def get_matrix(self):
        if self._invalid:
            transformed_coords = self._trans_shift.transform([[self._theta, 0]])[0]
            adjusted_theta = transformed_coords[0]
            rotation = transforms.Affine2D().rotate(adjusted_theta)
            self._mtx = IdentityTransform._mtx.copy()
            self._mtx[:2, :2] = rotation.get_matrix()[:2, :2]
        return self._mtx
```
The resulting graph has the radial caps shifted by $\pi/2$ counterclockwise from their correct positions and not properly rotated in place.

```python 
import matplotlib.pyplot as plt
import numpy as np
theta = np.arange(0, 2 * np.pi, np.pi / 8)
r = theta / np.pi / 2 + 0.5
fig = plt.figure(figsize=(5, 5))
ax = fig.add_subplot(projection='polar')
ax.errorbar(theta, r, xerr=0.1, yerr=0.1, capsize=7, fmt="o", c="seagreen")
ax.set_theta_zero_location("N")
ax.set_theta_direction(-1)
ax.set_yticklabels([])
plt.show()
```
<img width="495" alt="Screenshot 2024-10-12 at 3 54 09 PM" src="https://github.com/user-attachments/assets/dc471d2c-3949-4285-9982-f9d38ccea353">.

Reply:
Right. If we modify the `theta` inputs to `ScaledRotation`, we get
![my_plot-2](https://github.com/user-attachments/assets/44de9643-b855-4228-809e-f661622ee0a5)

I realized we can modify the output of the `ScaledRotation` instance in `errorbar` and still get dynamic updates. 

`errorbar`:
```python 
for theta, r in zip(l.get_xdata(), l.get_ydata()):
                        rotation = mtransforms.ScaledRotation(
                            theta=theta,
                            trans_shift = trans_shift)
                        if axis == 'y':
                            rotation += mtransforms.Affine2D().rotate(np.pi / 2)
                        ms = mmarkers.MarkerStyle(marker=marker,
                                                  transform=rotation)
```
`ScaledRotation`:
```python
class ScaledRotation(Affine2DBase):
    """
    A transformation that applies offset and direction
    based on *trans_shift*.
    """
    def __init__(self, theta, trans_shift):
        super().__init__()
        self._theta = theta
        self._trans_shift = trans_shift
        self._mtx = None

    def get_matrix(self):
        if self._invalid:
            transformed_coords = self._trans_shift.transform([[self._theta, 0]])[0]
            adjusted_theta = transformed_coords[0]
            rotation = transforms.Affine2D().rotate(adjusted_theta)
            self._mtx = IdentityTransform._mtx.copy()
            self._mtx[:2, :2] = rotation.get_matrix()[:2, :2]
        return self._mtx
```
Plotting:
```python 
import matplotlib.pyplot as plt
import numpy as np

theta = np.arange(0, 2 * np.pi, np.pi / 8)
r = theta / np.pi / 2 + 0.5
theta_zero_locations = ['N', 'NE', 'E', 'SE', 'S', 'SW', 'W', 'NW']
theta_directions = [1, -1]

fig, axs = plt.subplots(len(theta_directions), len(theta_zero_locations), 
                        figsize=(40, 20), subplot_kw={'projection': 'polar'})

for i, direction in enumerate(theta_directions):
    for j, zero_loc in enumerate(theta_zero_locations):
        ax = axs[i, j]
        ax.errorbar(theta, r, xerr=0.1, yerr=0.1, capsize=7, fmt="o", c="seagreen")
        ax.set_theta_zero_location(zero_loc)
        ax.set_theta_direction(direction)
        ax.set_yticklabels([])
        ax.set_xticklabels([])
        ax.set_title(f"Zero Loc: {zero_loc}\nDirection: {'CW' if direction == 1 else 'CCW'}", fontsize=10)

plt.tight_layout()
plt.savefig('my_plot.png')
plt.show()
```

![my_plot](https://github.com/user-attachments/assets/e60dfabc-c33f-4ecb-b3a6-708eedfd2e52)

Which means we don't have to pass the `ScaledRotation` instance directly to `mmarkers.MarkerStyle`.